src/kv/committable_tx.h

```diff
@@ -391,14 +391,21 @@ namespace kv
   // never conflict.
   class ReservedTx : public CommittableTx
   {
+  private:
+    Version rollback_count = 0;
+
   public:
     ReservedTx(
-      AbstractStore* _store, Term read_term, const TxID& reserved_tx_id) :
+      AbstractStore* _store,
+      Term read_term,
+      const TxID& reserved_tx_id,
+      Version rollback_count_) :
```

Comment:
Just a query: Can we use one of the existing `term` values here (either the `read_term` or `reserved_tx_id.term`), rather than passing an additional `rollback_count_`? Is `reserved_tx_id.term == store.commit_view()` the same as `store.check_rollback_count(expected_rollback_count)`?

Reply:
I think so, yes. There's not a lot of reason left to have rollback count now, as discussed, but if it stays it's appropriate for this use. I don't mind switching to term if we think it's better, but I don't want to bundle the removal of the rollback counter with this. And no, because the check must be under lock, otherwise we have a race condition.

Reply:
I agree it's not sensible to bundle _removal_ of the rollback counter into this change, but if we could avoid extending use of the rollback counter (passing as an additional arg here), that'd be nice.

I don't see the distinction if the check is made under a lock, because the action we take after the check is always _outside_ of the version lock (ie - any call into `store`, whether `check_rollback_count` or `commit_view`, might temporarily take `version_lock`, but if we do something _here_ then it'll be after `version_lock` is released) - we think that's safe because the map lock(s) prevent racing rollbacks.

Reply:
I hadn't noticed term_of_next_version was atomic, I agree it's equivalent.

> but if we could avoid extending use of the rollback counter (passing as an additional arg here), that'd be nice.

Here we're explicitly trying to detect a rollback though, and the rollback counter has just the right semantics and name for that, no more no less. We don't really care what term it is. I'm going to suggest this can be revisited separately if we ever get around to removing the rollback counter.